networkx/algorithms/distance_measures.py

```diff
@@ -769,3 +770,104 @@ def resistance_distance(G, nodeA, nodeB, weight=None, invert_weight=True):
     rd = Ldet * LdiagAB_s / LdiagA_s
 
     return rd
+
+
+@nx._dispatch(edge_attrs="weight")
+def kemeny_constant(G, weight=None):
+    """Returns Kemeny's constant of the given graph.
+
+    *Kemeny's constant* of a graph `G` can be computed by regarding the graph
+    as a Markov chain. Kemeny's constant is then the expected number of time steps
+    to transition from a starting state i to a random destination state
+    sampled from the Markov chain's stationary distribution.
+    Kemeny's constant is independent of the chosen initial state [1]_.
+
+    Kemeny's constant is used as a measure of connectivity of graphs.
+    Large values indicate a poorly connected graph whereas low values
+    correspond to well-connected graphs.
+
+    If weight is not provided, then a weight of 1 is used for all edges.
+
+    Parameters
+    ----------
+    G : NetworkX graph
+
+    weight : string or None, optional (default=None)
+       The edge data key used to compute Kemeny's constant.
+       If None, then each edge has weight 1.
+
+    Returns
+    -------
+    K : float
+        Kemeny's constant of the graph `G`.
+
+    Raises
+    ------
+    NetworkXError
+        If the graph `G` is directed.
+
+    NetworkXError
+        If the graph `G` is not connected.
+
+    NetworkXError
+        If the graph `G` does not contain any nodes.
+
+    NetworkXError
+        If the graph `G` contains edges with negative weights.
+
+    Examples
+    --------
+    >>> G = nx.complete_graph(5)
+    >>> round(nx.kemeny_constant(G), 10)
+    3.2
+
+    Notes
+    -----
+    The implementation is based on equation (3.3) in [2]_. Self-loops are included.
+    Multi-edges are contracted in one edge with weight equal to the sum of the weights.
```

Comment:
What does "self-loops included" refer to? 
Perhaps wording like:
"Self-loops are allowed and indicate a Markov Chain where
the state can remain the same. "